bodo/pandas/utils.py

```diff
@@ -329,18 +329,66 @@ def generate_duckdb(self, cache=None):
         if id(self) in cache:
             return cache[id(self)]
 
-        def recursive_check(x):
+        def recursive_check(x, in_filter):
             """Recursively convert LazyPlans but return other types unmodified."""
             if isinstance(x, LazyPlan):
-                return x.generate_duckdb(cache=cache)
+                return x.generate_duckdb(cache=cache, in_filter=in_filter)
             else:
                 return x
 
-        # Convert any LazyPlan in the args or kwargs.
-        args = [recursive_check(x) for x in self.args]
-        kwargs = {k: recursive_check(v) for k, v in self.kwargs.items()}
+        # Description of LogicalProjectionOrColRef node type.
```

Comment:
It'd be best if we could avoid this workaround. What is the specific example that is problematic? Let's discuss.

Reply:
Is my extensive comment there not clear?  Projection nodes take ownership of unique_ptr sources.  So, if you have two projection in one filter they both will try to claim ownership and the second one can't possibly get it because the source node (cython obj) that it knows about no longer has a reference to the owned unique_ptr.  It is just an impedance mismatch between the two APIs and yet another layer of lazy.

Reply:
Let's merge this for now. I'm wondering if we can extract the expression and avoid the logical operators with unique pointers.